### PR TITLE
feat: add trace option in `revme evm`

### LINKED ISSUE #1373: Add tracer option in `revme evm`
It would be great to have `--trace` cli option in revme evm command.

`revm evm 000000` command executed arbitrary bytecodes.

### PR DESCRIPTION
**Pull request by qiweiii**
resolve #1373

### CODE REVIEW

File: bins/revme/src/cmd/evmrunner.rs [hunk 1]
```diff
@@ -99,6 +104,25 @@ impl Cmd {
             if self.state {
                 println!("State: {:#?}", out.state);
             }
+            if self.trace {
```
Comment (rakita) [inline bins/revme/src/cmd/evmrunner.rs:107]: this `if self.trace {` should be moved few lines above.

File: bins/revme/src/cmd/evmrunner.rs [hunk 2]
```diff
@@ -99,6 +104,25 @@ impl Cmd {
...
+                println!("Traces:");
+                let mut evm = Evm::builder()
```
Comment (rakita) [inline bins/revme/src/cmd/evmrunner.rs:109]: You don't need to build a new Evm you can modify the existing one with `evm.modify().with_external_context(...).append_...().build()`

**Review by rakita — CHANGES_REQUESTED** [review]
good path, left few comments on refactor.

File: bins/revme/src/cmd/evmrunner.rs (hunk 1 above)
Comment (rakita) [reply bins/revme/src/cmd/evmrunner.rs:104]: Would put 102 line in `else` branch. If it is `--trace` we dont need to execute it twice.

And put line 103 println! after traces.

Comment (rakita) [reply bins/revme/src/cmd/evmrunner.rs:104]: @qiweii

File: bins/revme/src/cmd/evmrunner.rs [hunk 3]
```diff
@@ -93,9 +98,22 @@ impl Cmd {
...
+                .append_handler_register(inspector_handle_register)
+                .build();
+
+            let out = evm.transact().map_err(|_| Errors::EVMError)?;
+            println!("Result: {:#?}", out.result);
         } else {
             let out = evm.transact().map_err(|_| Errors::EVMError)?;
             println!("Result: {:#?}", out.result);
+
             if self.state {
```
Comment (rakita) [inline bins/revme/src/cmd/evmrunner.rs:117]: `self.trace` option should be independent of `self.trace`. So it should be put outside of else branch

File: bins/revme/src/cmd/evmrunner.rs [hunk 4]
```diff
@@ -93,9 +98,22 @@ impl Cmd {
             microbench::bench(&bench_options, "Run bytecode", || {
                 let _ = evm.transact().unwrap();
             });
+        } else if self.trace {
+            println!("Traces:");
```
Comment (rakita) [inline bins/revme/src/cmd/evmrunner.rs:102]:
```suggestion
```

There is no need to printout this

File: bins/revme/src/cmd/evmrunner.rs [hunk 5]
```diff
@@ -93,9 +98,22 @@ impl Cmd {
...
+            let mut evm = evm
+                .modify()
+                .reset_handler_with_external_context(TracerEip3155::new(
+                    Box::new(std::io::stdout()),
+                ))
+                .append_handler_register(inspector_handle_register)
+                .build();
+
+            let out = evm.transact().map_err(|_| Errors::EVMError)?;
+            println!("Result: {:#?}", out.result);
```
Comment (rakita) [inline bins/revme/src/cmd/evmrunner.rs:112]:
```suggestion
```
Or even this as the output of result is done inside `TracerEip3155`

File: bins/revme/src/cmd/evmrunner.rs (hunk 3 above)
Comment (qiweiii) [reply bins/revme/src/cmd/evmrunner.rs:117]: do you mean `independent of self.state`

File: bins/revme/src/cmd/evmrunner.rs [hunk 6]
```diff
@@ -93,13 +98,29 @@ impl Cmd {
...
+                .build();
+
+            evm.transact().map_err(|_| Errors::EVMError)?
         } else {
-            let out = evm.transact().map_err(|_| Errors::EVMError)?;
-            println!("Result: {:#?}", out.result);
-            if self.state {
-                println!("State: {:#?}", out.state);
-            }
+            evm.transact().map_err(|_| Errors::EVMError)?
```
Comment (rakita) [inline bins/revme/src/cmd/evmrunner.rs:116]:
```suggestion
            let out = evm.transact().map_err(|_| Errors::EVMError)?;
            println!("Result: {:#?}", out.result);
            out
```

File: bins/revme/src/cmd/evmrunner.rs [hunk 7]
```diff
@@ -93,13 +98,31 @@ impl Cmd {
...
         } else {
             let out = evm.transact().map_err(|_| Errors::EVMError)?;
             println!("Result: {:#?}", out.result);
-            if self.state {
-                println!("State: {:#?}", out.state);
-            }
+            out
+        };
+
+        println!("Result: {:#?}", out.result);
```
Comment (rakita) [inline bins/revme/src/cmd/evmrunner.rs:121]:
```suggestion
```

TracerEip3155 has its own output

**Review by rakita — APPROVED** [review]
lgtm.

Will merge after CI

**Pull request merged**